### ISSUE TITLE
rack update to 2.1.1 break rack-oauth2

### ISSUE THREAD

**Issue by utkarsh2102** [nov/rack-oauth2#75]
Hi,

https://github.com/nov/rack-oauth2/blob/767b973b0df80f4b31bf3e146d1ab4d5e087f584/rack-oauth2.gemspec#L16

As mentioned, of course, rack update to 2.1.1 breaks rack-oauth2 with the following test failures:
```
Failures:

  1) Rack::OAuth2::Server::Resource::BadRequest#finish should respond in JSON
     Failure/Error: response.body.should == ['{"error":"invalid_request"}']

     NoMethodError:
       undefined method `body' for ["{\"error\":\"invalid_request\"}"]:Array
     # ./spec/rack/oauth2/server/resource/error_spec.rb:13:in `block (3 levels) in <top (required)>'

  2) Rack::OAuth2::Server::Resource::Unauthorized when scheme is defined #finish should respond in JSON
     Failure/Error: response.body.should == ['{"error":"invalid_token"}']

     NoMethodError:
       undefined method `body' for ["{\"error\":\"invalid_token\"}"]:Array
     # ./spec/rack/oauth2/server/resource/error_spec.rb:47:in `block (4 levels) in <top (required)>'

  3) Rack::OAuth2::Server::Resource::Unauthorized when scheme is defined #finish when error_code is not invalid_token should have error_code in body but not in WWW-Authenticate header
     Failure/Error: response.body.first.should include '"error":"something"'

     NoMethodError:
       undefined method `body' for ["{\"error\":\"something\"}"]:Array
     # ./spec/rack/oauth2/server/resource/error_spec.rb:56:in `block (5 levels) in <top (required)>'

  4) Rack::OAuth2::Server::Resource::Unauthorized when scheme is defined #finish when no error_code is given should have error_code in body but not in WWW-Authenticate header
     Failure/Error: response.body.first.should == '{"error":"unauthorized"}'

     NoMethodError:
       undefined method `body' for ["{\"error\":\"unauthorized\"}"]:Array
     # ./spec/rack/oauth2/server/resource/error_spec.rb:66:in `block (5 levels) in <top (required)>'

  5) Rack::OAuth2::Server::Resource::Unauthorized when scheme is defined #finish when realm is specified should use given realm
     Failure/Error: response.body.first.should include '"error":"something"'

     NoMethodError:
       undefined method `body' for ["{\"error\":\"something\"}"]:Array
     # ./spec/rack/oauth2/server/resource/error_spec.rb:77:in `block (5 levels) in <top (required)>'

  6) Rack::OAuth2::Server::Resource::Forbidden#finish should respond in JSON
     Failure/Error: response.body.should == ['{"error":"insufficient_scope"}']

     NoMethodError:
       undefined method `body' for ["{\"error\":\"insufficient_scope\"}"]:Array
     # ./spec/rack/oauth2/server/resource/error_spec.rb:94:in `block (3 levels) in <top (required)>'

  7) Rack::OAuth2::Server::Resource::Forbidden when scope option is given should have blank WWW-Authenticate header
     Failure/Error: response.body.first.should include '"scope":"scope1 scope2"'

     NoMethodError:
       undefined method `body' for #<Array:0x00005580a289cb28>
     # ./spec/rack/oauth2/server/resource/error_spec.rb:103:in `block (3 levels) in <top (required)>'

  8) Rack::OAuth2::Server::Token::BadRequest#finish should respond in JSON
     Failure/Error: response.body.should == ['{"error":"invalid_request"}']

     NoMethodError:
       undefined method `body' for ["{\"error\":\"invalid_request\"}"]:Array
     # ./spec/rack/oauth2/server/token/error_spec.rb:13:in `block (3 levels) in <top (required)>'

  9) Rack::OAuth2::Server::Token::Unauthorized#finish should respond in JSON
     Failure/Error: response.body.should == ['{"error":"invalid_request"}']

     NoMethodError:
       undefined method `body' for ["{\"error\":\"invalid_request\"}"]:Array
     # ./spec/rack/oauth2/server/token/error_spec.rb:29:in `block (3 levels) in <top (required)>'

  10) Rack::OAuth2::Server::Token when multiple client credentials are given when different credentials are given should fail with unsupported_grant_type
      Failure/Error: response.body.first.should include '"error":"invalid_request"'

      NoMethodError:
        undefined method `body' for #<Array:0x00005580a1f30f08>
      # ./spec/rack/oauth2/server/token_spec.rb:33:in `block (4 levels) in <top (required)>'

Finished in 0.76006 seconds (files took 2.76 seconds to load)
374 examples, 10 failures

Failed examples:

rspec ./spec/rack/oauth2/server/resource/error_spec.rb:9 # Rack::OAuth2::Server::Resource::BadRequest#finish should respond in JSON
rspec ./spec/rack/oauth2/server/resource/error_spec.rb:42 # Rack::OAuth2::Server::Resource::Unauthorized when scheme is defined #finish should respond in JSON
rspec ./spec/rack/oauth2/server/resource/error_spec.rb:53 # Rack::OAuth2::Server::Resource::Unauthorized when scheme is defined #finish when error_code is not invalid_token should have error_code in body but not in WWW-Authenticate header
rspec ./spec/rack/oauth2/server/resource/error_spec.rb:63 # Rack::OAuth2::Server::Resource::Unauthorized when scheme is defined #finish when no error_code is given should have error_code in body but not in WWW-Authenticate header
rspec ./spec/rack/oauth2/server/resource/error_spec.rb:74 # Rack::OAuth2::Server::Resource::Unauthorized when scheme is defined #finish when realm is specified should use given realm
rspec ./spec/rack/oauth2/server/resource/error_spec.rb:90 # Rack::OAuth2::Server::Resource::Forbidden#finish should respond in JSON
rspec ./spec/rack/oauth2/server/resource/error_spec.rb:101 # Rack::OAuth2::Server::Resource::Forbidden when scope option is given should have blank WWW-Authenticate header
rspec ./spec/rack/oauth2/server/token/error_spec.rb:9 # Rack::OAuth2::Server::Token::BadRequest#finish should respond in JSON
rspec ./spec/rack/oauth2/server/token/error_spec.rb:24 # Rack::OAuth2::Server::Token::Unauthorized#finish should respond in JSON
rspec ./spec/rack/oauth2/server/token_spec.rb:30 # Rack::OAuth2::Server::Token when multiple client credentials are given when different credentials are given should fail with unsupported_grant_type
```

Any idea how to fix this?

**Comment by nov** [comment]
I made a fix for them, but reverted for now.
Are you already using rack 2.1+?

**Comment by utkarsh2102** [comment]
Hi @nov,
Yep, we need to update rack to 2.1.1.
Could you point me to the fix? Or give me a patch for it? So I can apply locally and build it for now?

**Comment by utkarsh2102** [comment]
Ah, I see. It was reverted in the latest version.
Could you perhaps help me patch this? Or something? Or any ETA about when will the release happen?

**Comment by ecylmz** [comment]
Hi @nov 

rack < 2.1.3 has this vulnerability: https://snyk.io/vuln/SNYK-RUBY-RACK-569066

We use the omniauth_openid_connect gem along with ruby on rails.

Omniauth_openid_connect gem gives dependence on this gem.

Ruby on Rails is forced to use version 2.0.9 of Rack and we cannot close the vulnerability if there is no rack version 2.1.3 or higher on this gem. 

Could you upgrade rack to version 2.1.3 or higher?

**Comment by nov** [comment]
I've just released `1.13.0.beta`.
Please try it.

**Comment by ecylmz** [comment]
It works for me. Thanks @nov

**Comment by nov** [comment]
Thanks, I've released `1.13.0`.
